internal/detector/detector.go

```diff
@@ -77,23 +80,25 @@ func (m *Marker) DetectFunctionalUpstream(us *v1.Upstream) (*v1.ServiceInfo, map
 	// try every possible detector concurrently
 	for _, d := range m.detectors {
 		go func(d Interface) {
```

Comment:
what bounds this goroutine? is it bounded by max retires? is there an guarantee that they dont run in parallel?

Reply:
cannot run in parallel for the same upstream, because we have one long-lived goroutine per upstream